applications/GeoMechanicsApplication/custom_utilities/generic_utilities.h

```diff
@@ -48,6 +49,36 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) GenericUtilities
         return result;
     }
 
+    template <class ContainerType, class IdsType>
+    static void CollectIdsFromEntity(const ContainerType& rContainer, IdsType& rIds)
+    {
+        for (const auto& r_item : rContainer) {
+            const auto id = r_item.Id();
+
+            if constexpr (requires { rIds.insert(id); }) {
+                // Works for set/unordered_set
+                rIds.insert(id);
+            } else if constexpr (requires { rIds.push_back(id); }) {
+                // Works for vector/deque
+                rIds.push_back(id);
+            } else {
+                static_assert(!std::is_same_v<IdsType, IdsType>,
+                              "OutputContainerType does not support insert or push_back.");
+            }
+        }
+    }
+
+    template <class ContainerType>
+    static auto CollectIdsFromEntity(const ContainerType& rContainer)
+    {
+        using ItemRef = decltype(*std::begin(rContainer));
+        using Item    = std::remove_cv_t<std::remove_reference_t<ItemRef>>;
+        using IdType  = decltype(std::declval<Item>().Id());
+        std::vector<IdType> result;
+        CollectIdsFromEntity(rContainer, result);
```

Comment:
When changing the previous function to accept an output iterator, this would change to: 
```suggestion
        result.reserve(rContainer.size());
        CollectIdsFromEntity(rContainer, std::back_inserter(result));
```
I also added a reserve for performance